src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/MappingExclusions.kt

```diff
@@ -16,3 +16,6 @@ annotation class ExcludeJpaManagedFieldsIncludingDisplayNameFields
 
 @Mapping(target = "reference", ignore = true)
 annotation class ExcludeReferenceField
+
+@Mapping(target = "reference", expression = "java(UUID.randomUUID())")
+annotation class GenerateNewReference
```

Comment:
Nice 👍

Reply:
I thought you'd like that 👍 😁

Reply:
I do. I did consider the filename, but it's fine!